### DISCUSSION TITLE
'Format' a property in Read only object

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3367] [Questions]
I have to 'format' a property in a ReadOnly object.
I know ReadOnlyBase has AddBusinessRules method which is used for authorization rules.
But is there a way to 'format' a property in readonly object? Essentially I have to split a property name into multiple lines based on length of the value.

Regards

**Comment by rockfordlhotka** [comment]
Normally you'd format the value as you load it from the database. Or format it in the UI for display purposes.

**Reply by Chicagoan2016** [reply]
I changed the collection to read-write since we already have the 'Format' business rule(s).
We needed this RO collection as a data source for rdlc reports.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka , is it possible to call 'format' method  from GetProperty? (readonly object).

**Reply by rockfordlhotka** [reply]
It should be, but you might need to surpress an analyzer warning.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka I just thought of something I would like to share with you :)
Maybe we could provide a custom GetPropertyConvert in 'conversion' properties?

**Reply by rockfordlhotka** [reply]
Maybe?

Usually the UI framework provides some sort of formatting or conversion mechanism. It is quite formal in XAML for example, with value converters. 

Is it really a business rule to reformat text?

Maybe it is?
